src/extract.js

```diff
@@ -462,6 +464,27 @@ exports.Extractor = class Extractor {
       }, []);
   }
 
+  _getNodeHTML(node) {
+    const html = node.html();
+
+    if (this.options.removeHTMLWhitespaces === true) {
+      return html
+                 // From https://medium.com/@patrickbrosset/when-does-white-space-matter-in-html-b90e8a7cdd33
+                 // All spaces and tabs immediately before and after a line break are ignored
+                 .replace(/\s+\n/g, '\n')
+                 .replace(/\n\s+/g, '\n')
+                 // All tab characters are handled as space characters
+                 .replace(/\t/g, ' ')
+                 // Line breaks are converted to spaces:
+                 .replace(/\n/g, ' ')
+                 // Any space immediately following another space (even across two separate inline elements) is ignored
+                 .replace(/ +/g, ' ')
+                 .trim();
```

Comment:
nit: I would rather using the same whitespace spacing as everywhere else in the code :) 2 spaces !